cliquet/storage/postgresql/__init__.py

```diff
@@ -586,9 +586,13 @@ def _format_conditions(self, filters, id_field, modified_field,
                 # Safely escape field name
                 field_holder = '%s_field_%s' % (prefix, i)
                 holders[field_holder] = filtr.field
+
                 # JSON operator ->> retrieves values as text.
                 # If field is missing, we default to ''.
                 sql_field = "coalesce(data->>:%s, '')" % field_holder
+                if isinstance(value, (int, float)) and \
+                   value not in (True, False):
+                    sql_field = "(data->>:%s)::numeric" % field_holder
```

Comment:
Much better than the previous version!

@Natim it makes me think: couldn't we have converted the `value` to string instead? (*not for sorting I know*)